compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposePanel.desktop.kt

```diff
@@ -207,19 +206,12 @@ class ComposePanel @ExperimentalComposeUiApi constructor(
             _focusListeners.forEach(contentComponent::addFocusListener)
             contentComponent.addFocusListener(object : FocusListener {
                 override fun focusGained(e: FocusEvent) {
-                    // The focus can be switched from the child component inside SwingPanel.
-                    // In that case, SwingPanel will take care of it.
-                    if (!isParentOf(e.oppositeComponent)) {
-                        focusManager.requestFocus()
+                    if (!e.isTemporary && !e.isFocusGainedHandledBySwingPanel(this@ComposePanel)) {
                         when (e.cause) {
-                            FocusEvent.Cause.TRAVERSAL_FORWARD -> {
```

Comment:
This is moved to `ComposeSceneMediator`, as it is valid for all containers using it